src/core/Akka/Dispatch/Dispatchers.cs

```diff
@@ -186,84 +147,211 @@ public MessageDispatcher DefaultGlobalDispatcher
+            get
+            {
+                return
+                    IdConfig(DefaultDispatcherId)
+                        .WithFallback(Prerequisites.Settings.Config.GetConfig(DefaultDispatcherId));
+            }
+        }
+
+        /// <summary>
+        /// INTERNAL API
+        /// 
+        /// Used when a plugin like Akka.Persistence needs to be able to load dispatcher configurations to the chain.
+        /// </summary>
+        internal void ReloadPrerequisites(IDispatcherPrerequisites prerequisites)
+        {
+            Prerequisites = prerequisites;
+            _cachingConfig = new CachingConfig(prerequisites.Settings.Config);
         }
 
+        /// <summary>
+        /// The prerequisites required for some <see cref="MessageDispatcherConfigurator"/> instances.
+        /// </summary>
+        public IDispatcherPrerequisites Prerequisites { get; private set; }
 
+        /// <summary>
+        /// Returns a dispatcher as specified in configuration. Please note that this method _MAY_
+        /// create and return a new dispatcher on _EVERY_ call.
+        /// </summary>
+        /// <exception cref="ConfigurationException">If the specified dispatcher cannot be found in configuration.</exception>
         public MessageDispatcher Lookup(string dispatcherName)
         {
-            return FromConfig(dispatcherName);
+            return LookupConfigurator(dispatcherName).Dispatcher();
         }
 
         /// <summary>
-        ///     Froms the configuration.
+        /// Checks that configuration provides a sectionfor the given dispatcher.
+        /// This does not gaurantee that no <see cref="ConfigurationException"/> will be thrown
+        /// when using the dispatcher, because the details can only be checked by trying to
+        /// instantiate it, which might be undersirable when just checking.
         /// </summary>
-        /// <param name="path">The path.</param>
-        /// <returns>MessageDispatcher.</returns>
-        public MessageDispatcher FromConfig(string path)
+        public bool HasDispatcher(string id)
         {
-            //TODO: this should not exist, it is only here because we dont serialize dispathcer when doing remote deploy..
-            if (string.IsNullOrEmpty(path))
+            return _dispatcherConfigurators.ContainsKey(id) || _cachingConfig.HasPath(id);
+        }
+
+        private MessageDispatcherConfigurator LookupConfigurator(string id)
+        {
+            MessageDispatcherConfigurator configurator;
+            if (!_dispatcherConfigurators.TryGetValue(id, out configurator))
             {
-                var disp = new ThreadPoolDispatcher
+                // It doesn't matter if we create a dispatcher configurator that isn't used due to concurrent lookup.
+                // That shouldn't happen often and in case it does the actual ExecutorService isn't
+                // created until used, i.e. cheap.
```

Comment:
I think so but all of the funky error handling and lookup stuff would have to be done inside a lambda. So I ran with this just because it was easier to debug.